core/capabilities/ccip/ccipevm/gas_helpers.go

```diff
@@ -44,7 +48,7 @@ func (gp EstimateProvider) CalculateMerkleTreeGas(numRequests int) uint64 {
 		return 0
 	}
 	merkleProofBytes := (math.Ceil(math.Log2(float64(numRequests))))*32 + (1+2)*32 // only ever one outer root hash
-	return uint64(merkleProofBytes * CalldataGasPerByte)
+	return uint64(merkleProofBytes * CalldataGasPerByteBase)
```

Comment:
q: why not align this with the onchain calc? Would make for more consistent reading of the code

```golang
func eip7623(numBytes int) int {
  destCallDataCost := numBytes * CalldataGasPerByteBase;
  if (numBytes > CalldataGasPerByteThreshold) {
    destCallDataCost = CalldataGasPerByteBase * CalldataGasPerByteThreshold
        + (numBytes - CalldataGasPerByteThreshold) * CalldataGasPerByteHigh;
    }
	return destCallDataCost
}

// old code: return uint64(merkleProofBytes * CalldataGasPerByteBase)
return uint64(eip7623(merkleProofBytes))
```

I guess one could argue that this amount of gas is not really needed, but it was basically what was paid for. Either way I'm kinda indifferent.

Reply:
Discussed offline 👍🏼